src/boot/ast.ml

```diff
@@ -172,7 +174,7 @@ and seqMatchType =
 (* Patterns *)
 and pat =
 | PatNamed of info * patName                      (* Named, capturing wildcard *)
-| PatSeq   of info * pat list * seqMatchType      (* Sequence pattern *)
+| PatSeq   of info * pat Mseq.t * seqMatchType    (* Sequence pattern *)
```

Comment:
Should this really be an `Mseq.t`? It seems like this is representing a piece of MExpr syntax and not a runtime value.

Reply:
The main reason for this data-structure is that we apply a `fold_right2` over a sequence of patterns and a sequence when matching over sequences in `mexpr.ml` line 518. If we keep these patterns in a list we would have to write a custom `fold_right2` that folds over a sequence and a list rather than two sequences.

Reply:
Idk what is best

Reply:
That sounds reasonable!

Reply:
I thought about it some more, I think we can change it to a list and then just convert it to a sequence when doing a matching. Since the number of patterns will only be "a handful", speed should not be a factor.